activerecord/lib/active_record/errors.rb

```diff
@@ -185,6 +185,21 @@ class PreparedStatementInvalid < ActiveRecordError
 
   # Raised when a given database does not exist.
   class NoDatabaseError < StatementInvalid
+    include ActiveSupport::ActionableError
+
+    action "Create database" do
+      ActiveRecord::Tasks::DatabaseTasks.create_current
```

Comment:
This would only create the default database. The `NoDatabaseError` could be thrown for a secondary database.

Reply:
Sounds good, will make that update.

Reply:
@tgxworld I tested this out and it seems to be working for me, here is the configuration I used:

```
development:
  primary:
    database: my_primary_database
    adapter: postgresql
  primary_replica:
    database: my_primary_database
    replica: true
    adapter: postgresql
  animals:
    database: my_animals_database
    migrations_paths: db/animals_migrate
    adapter: postgresql
  animals_replica:
    database: my_animals_database
    replica: true
    adapter: postgresql

# test db
test:
  <<: *default
  database: askaway_test
```

Here is the output from the console, showing I created 3 databases:
```
Started POST "/rails/actions" for 127.0.0.1 at 2020-07-09 16:50:11 +0100
   (80.8ms)  CREATE DATABASE "my_primary_database" ENCODING = 'utf8'
  ↳ /Users/haroon/projects/rails/activerecord/lib/active_record/log_subscriber.rb:109:in `debug'
Created database 'my_primary_database'
   (78.7ms)  CREATE DATABASE "my_animals_database" ENCODING = 'utf8'
  ↳ /Users/haroon/projects/rails/activerecord/lib/active_record/log_subscriber.rb:109:in `debug'
Created database 'my_animals_database'
   (80.5ms)  CREATE DATABASE "askaway_test" ENCODING = 'unicode'
  ↳ /Users/haroon/projects/rails/activerecord/lib/active_record/log_subscriber.rb:109:in `debug'
Created database 'askaway_test'
```

Reply:
If you can let me know what configs you used to repro, I can try and fix... 🙂

Reply:
Sorry you're right. I just realized that when the error is thrown, the active connection_pool will be that of the missing database.